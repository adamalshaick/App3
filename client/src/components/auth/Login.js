import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { loginUser } from "../../actions/authActions";
import { Link } from "react-router-dom";
import InputGroup from "../common/InputGroup";
import Navbar from "../layout/Navbar";
import handleInputErrors from "../common/hoc/handleInputErrors";
import redirectAuthenticated from "../common/hoc/redirectAuthenticated";
import { ShadowCard } from "../common/styles/ShadowCard";

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: ""
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();
    const userData = {
      email: this.state.email,
      password: this.state.password
    };
    this.props.loginUser(userData);
  };
  render() {
    const { errors } = this.props;

    return (
      <>
        <Navbar />
        <div className="container">
          <div className="row mt-5 entry">
            <div className="col-md-2 col-lg-3" />
            <ShadowCard className="col-md-8 col-lg-6 card mt-5 text-center p-0 ">
              <div className="card-body p-5">
                <h3 className="mb-5">Log in to your account</h3>
                <form noValidate onSubmit={this.onSubmit}>
                  <InputGroup
                    placeholder="Email Adress"
                    name="email"
                    type="email"
                    value={this.state.email}
                    onChange={this.onChange}
                    error={errors.email}
                    id="email"
                    icon={"far fa-envelope"}
                  />
                  <InputGroup
                    placeholder="Password"
                    name="password"
                    type="password"
                    value={this.state.password}
                    onChange={this.onChange}
                    error={errors.password}
                    id="password"
                    icon={"fas fa-unlock"}
                  />
                  <button
                    type="submit"
                    className="btn btn-outline-primary btn-block mt-4"
                  >
                    Log In
                  </button>
                </form>
              </div>
              <div className="text-muted card-footer mt-3">
                Don't have an account yet? <Link to="/register">Sign Up</Link>
              </div>
            </ShadowCard>
          </div>
        </div>
      </>
    );
  }
}

Login.propTypes = {
  loginUser: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired
};

export default connect(
  null,
  { loginUser }
)(redirectAuthenticated(handleInputErrors(Login)));
