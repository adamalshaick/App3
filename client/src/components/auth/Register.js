import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { registerUser } from "../../actions/authActions";
import { Link } from "react-router-dom";
import Navbar from "../layout/Navbar";
import InputGroup from "../common/InputGroup";
import handleInputErrors from "../common/hoc/handleInputErrors";
import redirectAuthenticated from "../common/hoc/redirectAuthenticated";
import { ShadowCard } from "../common/styles/ShadowCard";

export class Register extends Component {
  constructor() {
    super();
    this.state = {
      name: "",
      email: "",
      password: "",
      password2: ""
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  onSubmit = e => {
    e.preventDefault();

    const newUser = {
      name: this.state.name,
      email: this.state.email,
      password: this.state.password,
      password2: this.state.password2
    };

    this.props.registerUser(newUser, this.props.history);
  };

  render() {
    const { errors } = this.props;
    return (
      <>
        <Navbar />
        <div className="container">
          <div className="row mt-5 entry">
            <div className="col-md-2 col-lg-3" />
            <ShadowCard className="col-md-8 col-lg-6 card mt-5 text-center p-0 ">
              <div className="card-body p-5">
                <h3 className="mb-5">Sign up</h3>
                <form noValidate onSubmit={this.onSubmit}>
                  <InputGroup
                    id="name"
                    placeholder="Name"
                    name="name"
                    value={this.state.name}
                    onChange={this.onChange}
                    error={errors.name}
                    icon={"fas fa-signature"}
                  />
                  <InputGroup
                    id="email"
                    placeholder="Email"
                    name="email"
                    type="email"
                    value={this.state.email}
                    onChange={this.onChange}
                    error={errors.email}
                    icon={"far fa-envelope"}
                  />
                  <InputGroup
                    id="password"
                    placeholder="Password"
                    name="password"
                    type="password"
                    value={this.state.password}
                    onChange={this.onChange}
                    error={errors.password}
                    icon={"fas fa-unlock"}
                  />
                  <InputGroup
                    id="password2"
                    placeholder="Confirm Password"
                    name="password2"
                    type="password"
                    value={this.state.password2}
                    onChange={this.onChange}
                    error={errors.password2}
                    icon={"fas fa-unlock"}
                  />
                  <button
                    type="submit"
                    className="btn btn-outline-primary btn-block mt-4"
                  >
                    Create Account
                  </button>
                </form>
              </div>
              <div className="text-muted card-footer mt-3">
                Already have an account? <Link to="/login">Log in</Link>
              </div>
            </ShadowCard>
          </div>
        </div>
      </>
    );
  }
}

Register.propTypes = {
  registerUser: PropTypes.func.isRequired,
  errors: PropTypes.object.isRequired
};

export default connect(
  null,
  { registerUser }
)(handleInputErrors(redirectAuthenticated(Register)));
