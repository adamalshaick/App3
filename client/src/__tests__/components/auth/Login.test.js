import React from "react";
import { Login } from "../../../components/auth/Login";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const mockLoginfn = jest.fn();
const Props = {
  errors: {}
};
let wrapper;

wrapper = shallow(<Login {...Props} loginUser={mockLoginfn} />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe("login action", () => {
  wrapper
    .find("#email")
    .simulate("change", { target: { name: "email", value: "test@gmail.com" } });

  wrapper.find("#password").simulate("change", {
    target: { name: "password", value: "passwordTest" }
  });

  it("should call the mock login function", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(mockLoginfn.mock.calls.length).toEqual(1);
  });

  it("calls function with correct data", () => {
    expect(mockLoginfn.mock.calls[0][0]).toEqual({
      email: "test@gmail.com",
      password: "passwordTest"
    });
  });
});
