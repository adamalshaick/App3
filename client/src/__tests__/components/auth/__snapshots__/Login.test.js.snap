// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<Fragment>
  <Connect(Navbar) />
  <div
    className="container"
  >
    <div
      className="row mt-5 entry"
    >
      <div
        className="col-md-2 col-lg-3"
      />
      <styled.div
        className="col-md-8 col-lg-6 card mt-5 text-center p-0 "
      >
        <div
          className="card-body p-5"
        >
          <h3
            className="mb-5"
          >
            Log in to your account
          </h3>
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <InputGroup
              icon="far fa-envelope"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="Email Adress"
              type="email"
              value="test@gmail.com"
            />
            <InputGroup
              icon="fas fa-unlock"
              id="password"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value="passwordTest"
            />
            <button
              className="btn btn-outline-primary btn-block mt-4"
              type="submit"
            >
              Log In
            </button>
          </form>
        </div>
        <div
          className="text-muted card-footer mt-3"
        >
          Don't have an account yet? 
          <Link
            replace={false}
            to="/register"
          >
            Sign Up
          </Link>
        </div>
      </styled.div>
    </div>
  </div>
</Fragment>
`;
